testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -163,15 +165,18 @@ abstract class HttpClientTest extends InstrumentationSpecification {
     given:
     assumeTrue(testCallbackWithParent())
 
+    def status = new BlockingVariable<Integer>()
```

Comment:
I started with this PR so the test framework is in charge of blocking, not the request executors. Now the latter can be switched to async since they don't need to block.